core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt

```diff
@@ -264,9 +274,14 @@ internal class CBPeripheralCoreBluetoothPeripheral(
             centralManager.discoverServices(cbPeripheral, servicesToDiscover)
         }
 
-        cbPeripheral.services?.forEach { cbService ->
+        cbPeripheral.services?.filterIsInstance<CBService>()?.forEach { cbService ->
```

Comment:
Is the `filterIsInstance` so that we don't have to do the cast below?

Despite this being cleaner code, I think I'd still prefer the "unsafe" cast as it avoids an object allocation that `filterIsInstance` incurs.

```suggestion
        // Cast should be safe since `CBPeripheral.services` type is `[CBService]?`, according to:
        // https://developer.apple.com/documentation/corebluetooth/cbperipheral/services
        val discoveredServices = cbPeripheral.services as List<CBService>?
        discoveredServices?.forEach { cbService ->
```

Reply:
yes I've used filterIsInstance because I needed cbService casted to `CBService ` also in the next code block and didn't want to double cast it.

I'll push an update to the PR with the approach you suggested